File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        op = [1] * len(nums)

        prefix = 1
        for i in range(len(nums)):
            op[i] = prefix
            prefix *= nums[i]
        
        postfix = 1
        for i in range(len(nums) - 1, -1, -1):
            op[i] *= postfix
            postfix *= nums[i]

        return op 

        

File: Data Structures & Algorithms/rotate-array/submission-0.java
class Solution {
    public void rotate(int[] nums, int k) {
        int n = nums.length;
        k = k % n;

        while (k > 0) {
            int tp = nums[n - 1];
            for (int i = n - 1; i > 0; i--) {
                nums[i] = nums[i - 1];
            }
            nums[0] = tp;
            k--;
        }
    }
}